base/reducedim.jl

```diff
@@ -57,24 +57,33 @@ function reduced_indices0(inds::Indices{N}, region) where N
     tuple(rinds...)::typeof(inds)
 end
 
-
 ###### Generic reduction functions #####
 
+## extensions to the standard functions for BigInt, String and others
+
+zero2(t::Type) =  zero(t)
+zero2(::Type{T}) where T<:Number = zero(T)
+zero2(::Type{T}) where T<:AbstractString = ""
```

Comment:
As long as this isn't exported, it's not a feature addition. It just seems fishy to need a function like zero but for which `""` is the answer for strings. (Nevermind that `""` is a reasonable 1 value for strings; there is no zero value unless you embed strings into regular patterns, in which case the empty pattern is the zero value – but it's not a string.)